RestaurantReservation.API/Controllers/OrdersController.cs

```diff
@@ -0,0 +1,132 @@
+﻿using Asp.Versioning;
+using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using RestaurantReservation.API.DomainErrors;
+using RestaurantReservation.API.Models.MenuItems;
+using RestaurantReservation.API.Models.Orders;
+using RestaurantReservation.Db.Entities;
+using RestaurantReservation.Db.Repositories;
+using System.Text.Json;
+
+namespace RestaurantReservation.API.Controllers
+{
+    /// <summary>
+    /// Managing orders controller
+    /// </summary>
+    [Route("api/orders")]
+    [Authorize]
+    [ApiVersion("1.0")]
+    [ApiController]
+    public class OrdersController : ControllerBase
+    {
+        private readonly OrderRepository _orderRepository;
+        private readonly IMapper _mapper;
+        private const int DefaultPageSize = 10;
+
+        /// <summary>
+        /// Constructor for OrdersController
+        /// </summary>
+        /// <param name="orderRepository">The repository for order data</param>
+        /// <param name="mapper">The AutoMapper instance</param>
+        /// <exception cref="ArgumentNullException">Thrown if orderRepository or mapper is null</exception>
+        public OrdersController(OrderRepository orderRepository, IMapper mapper)
+        {
+            _orderRepository = orderRepository ?? throw new ArgumentNullException(nameof(orderRepository));
+            _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+        }
+
+
+        /// <summary>
+        /// Get list of orders
+        /// </summary>
+        /// <param name="pageNumber">The page number to return</param>
+        /// <param name="pageSize">Max number of records per page</param>
+        /// <returns>List of orders</returns>
+        /// <response code="200">Returns the list of orders</response>
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<OrderWithoutDetailsDto>>> GetOrders(int pageNumber = 1, int pageSize = DefaultPageSize)
+        {
+            var (orderEntities, paginationMetaData) = await _orderRepository.GetAllAsync(pageNumber, pageSize);
+            Response.Headers.Add("X-Pagination", JsonSerializer.Serialize(paginationMetaData));
+            return Ok(_mapper.Map<IEnumerable<OrderWithoutDetailsDto>>(orderEntities));
+        }
+
+        /// <summary>
+        /// Get order by Id
+        /// </summary>
+        /// <param name="id">Id of order to return</param>
+        /// <returns>Order</returns>
+        /// <response code="200">Returns the order with the specified id</response>
```

Comment:
isn't better to use the attributes here to indicate the possible returned responses